impl/src/main/java/com/okta/sdk/impl/ds/DefaultDataStore.java

```diff
@@ -470,14 +475,24 @@ private Response execute(Request request) throws ResourceException {
         Map<String, Object> out = null;
 
         if (response.hasBody()) {
-            out = mapMarshaller.unmarshal(response.getBody(), response.getHeaders().getLinkMap());
+            if ("json".equals(response.getHeaders().getContentType().getSubtype())) {
+                out = mapMarshaller.unmarshal(response.getBody(), response.getHeaders().getLinkMap());
+            } else {
+                try (InputStreamReader inputStreamReader = new InputStreamReader(response.getBody(), Charsets.UTF_8)) {
+                    out = ImmutableMap.of(MediaType.TEXT_PLAIN_VALUE, CharStreams.toString(inputStreamReader));
```

Comment:
IMHO, I feel this overloads this functionality in a confusing way.  If we need to expose raw HTTP related functionality, then new methods should be added for that purpose, i.e. `client.httpRaw().post("/url/")`

_Ideally_, we want to return an input stream to avoid large strings in memory (when possible)

Reply:
I don't quite understand why we need new methods. The aim of this PR - is to avoid exceptions of converting different formats to Map in the process of `getResource` method https://github.com/okta/okta-sdk-java/blob/90436758263d2c597319838af77cb848b2ae6f11/impl/src/main/java/com/okta/sdk/impl/ds/DefaultDataStore.java#L175
I've updated my code to return `Collections.singletonMap(MediaType.TEXT_PLAIN_VALUE, response.getBody())`, if the content type other than json.

Reply:
The SDKs original purpose was to hide all of the HTTP logic, so the user only deals with resources (and all of the complexities that come with that, like JSON objects changing over time).

Returning a string in a map isn't a resource, IMHO, sticking the response body in a map isn't intuitive.